### PR TITLE
@W-9282057 align compiler config with distribution config

### PR DESCRIPTION
**Pull request by lizaiv77**
- Classes Module, Namespace, Profile were moved to utam-compiler module to class JsonCompilerConfig
- JsonBasedCompilerConfig renamed to JsonCompilerConfig 
- JsonBasedLoaderConfig renamed to JsonLoaderConfig

matching PR https://git.soma.salesforce.com/LPOP/utam-java-framework/pull/889

example of compiler config
```json
{
  "name": "baseLwc",
  "pageObjectsDirectory": "/src/main/resources/spec",
  "namespaces" : [
    {
      "typePrefix" : "utam-lightning",
      "pathMatchRegex": ".*/lightning"
    },
    {
      "typePrefix" : "utam-test",
      "pathMatchRegex": ".*/helpers"
    },
    {
      "typePrefix" : "utam-aura",
      "pathMatchRegex": ".*/ui"
    }
  ],
  "target": {
    "pageObjects": "/src/main/java/pageObjects",
    "resources": "/src/main/resources",
    "unitTests": "/src/test/java",
    "unitTestsRunner": "TESTNG"
  }
}
```

### CODE REVIEW

File: utam-compiler/src/test/java/utam/compiler/translator/JsonCompilerConfigTests.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import utam.core.framework.consumer.UtamError;
+import utam.core.framework.context.StringValueProfile;
+
+/**
+ * tests for json config
+ *
+ * @author elizaveta.ivanova
+ * @since 234
+ */
+public class JsonCompilerConfigTests {
```
Comment (lizaiv77) [inline utam-compiler/src/test/java/utam/compiler/translator/JsonCompilerConfigTests.java:51]: this is not new file, simply renamed JsonBasedCompilerConfigTests.java

File: utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+import utam.core.declarative.translator.UnitTestRunner;
+
+/**
+ * JSON based config for UTAM compiler, class is public because of inner classes that can be used in
+ * downstream projects
+ *
+ * @author elizaveta.ivanova
+ * @since 234
+ */
+public class JsonCompilerConfig {
```
Comment (lizaiv77) [inline utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java:39]: this is not new file, renamed JsonBasedCompilerConfigTests.java

File: utam-core/src/main/java/utam/core/framework/consumer/JsonLoaderConfig.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import utam.core.framework.UtamCoreError;
+import utam.core.framework.context.StringValueProfile;
+
+/**
+ * JSON config for UTAM Loader public because used to write config from distribution module
+ *
+ * @author elizaveta.ivanova
+ * @since 234
+ */
+public class JsonLoaderConfig {
```
Comment (lizaiv77) [inline utam-core/src/main/java/utam/core/framework/consumer/JsonLoaderConfig.java:32]: not new file, renamed JsonBasedLoaderConfig.java

File: utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+     * Initializes a new instance of the Namespace class. Instantiated via JSON deserialization.
+     *
+     * @param typePrefix     type prefix
+     * @param pathMatchRegex the mapped name used by the UTAM compiler to generate packages for the
+     *                       generated Page Object source code
+     */
+    @JsonCreator
+    public Namespace(
+        @JsonProperty(value = "typePrefix", required = true) String typePrefix,
+        @JsonProperty(value = "pathMatchRegex", required = true) String pathMatchRegex) {
```
Comment (lizaiv77) [inline utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java:281]: @jimevanssfdc let me know if you're ok with property names for namespace

File: utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java (hunk 2 above)
Comment (jimevanssfdc) [reply utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java:39]: Understood. Approved for this file.

File: utam-compiler/src/main/java/utam/compiler/translator/TranslatorGenerationCommand.java [hunk 5]
```diff
@@ -43,13 +43,18 @@
   static final String OUTPUT_DIRECTORY_MISSING = "Output directory is not configured";
   static final String PACKAGE_CONFIG_MISSING = "Packages mapping is not configured";
   static final String REDUNDANT_CLI_ARGS = "If JSON file is set, all other arguments are ignored";
+  static final String ERR_COMPILER_CONFIG_NEEDS_ROOT = "To resolve configuration paths, set compilerRoot";
 
-  @Option(names = {"-c", "-config", "--config"},
+  @Option(names = {"-g", "-config", "--config"},
```
Comment (jimevanssfdc) [inline utam-compiler/src/main/java/utam/compiler/translator/TranslatorGenerationCommand.java:48]: Maybe we make this option `--configFile` with a shortcut of `-f`, and change the `--compilerRoot` shortcut to `-c`? It's early enough in this feature we can change it without hurting anything. Later on, we'll probably want to remove the other configuration command-line options.

File: utam-compiler/src/test/java/utam/compiler/translator/JsonCompilerConfigTests.java (hunk 1 above)
Comment (jimevanssfdc) [reply utam-compiler/src/test/java/utam/compiler/translator/JsonCompilerConfigTests.java:51]: Understood. Approved for this file.

File: utam-core/src/main/java/utam/core/framework/consumer/JsonLoaderConfig.java (hunk 3 above)
Comment (jimevanssfdc) [reply utam-core/src/main/java/utam/core/framework/consumer/JsonLoaderConfig.java:26]: Approved for this file.

**Review by jimevanssfdc — CHANGES_REQUESTED** [review]
Good work here. Just a minor nitpick or two, but I'm largely happy with it.

File: utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java (hunk 4 above)
Comment (jimevanssfdc) [reply utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java:289]: I'm entirely agnostic on property names. These seem perfectly fine to me. They're descriptive and their purpose is clear.

File: utam-compiler/src/main/java/utam/compiler/translator/TranslatorGenerationCommand.java (hunk 5 above)
Comment (lizaiv77) [reply utam-compiler/src/main/java/utam/compiler/translator/TranslatorGenerationCommand.java]: done and done. thx.

**Comment by lizaiv77** [comment]
@jimevanssfdc all comments addressed, ready for final review

**Review by jimevanssfdc — APPROVED** [review]

**Pull request merged**